### ISSUE TITLE
Pretrain PEGASUS from scratch

### ISSUE THREAD

**Issue by EKebriaei** [google-research/pegasus#126]
I want to pre-train PEGASUS model from scratch on a language other than English. Is there any way to do this using huggingace API's? Could anyone help with the pipeline? I tried to compile tensor ops but many linker errors I got. Please describe the pretraining steps (which .cc files to compile, how to do it and how to pretrain using these files).

**Comment by demegire** [comment]
+1

**Comment by EKebriaei** [comment]
I tried to pre-train PEGASUS on another language by the following steps:
1- I implemented pretrain-parsing-ops.cc and did not use the code in the repo (Because of compilation errors). 
2- Created my vocab file using sentencepieceBPE.
3- Gave the documents to the code (my pretrain-parsing-ops) and got inputs and targets. Each input row is a document in which some sentences are removed and replaced by <mask> token (According to probabilities for masking and rouge calculations). Each target row consists of sentences removed from the corresponding document. 
4- Change my_dataset.py to read from my input and target from the 3rd step. 
5- Running train.py without setting checkpoint switch as follow:
```
!python pegasus/bin/train.py --params=my_param \
--param_overrides=vocab_filename=/root/pegasus/ckpt/fr_bpe.model \
--model_dir=output
```
I run it over GPU and batch_size is 3. After 200K steps in evaluation phase, I got wiered predictions. Almost 90% of predictions consist of a repetition of a word or phrase! 
When I did this with `--train_init_checkpoint=ckpt/model.ckpt-1500000` (Using pre-trained checkpoint), I got better predictions on the data (I did fine-tuning over a language other than English with checkpoints of main PEGASUS which obtained from pretraining over English datasets C4 & CC).
Can anyone help me with this? Am I correct in doing pertaining? 

>

**Comment by JingqingZ** [comment]
200k step with batch size 3 is probably too early to conclude as the model has not observed much data yet. As a reference, PEGASUS was pre-trained with batch size 8192 for 1.5M steps.

**Comment by EKebriaei** [comment]
> 200k step with batch size 3 is probably too early to conclude as the model has not observed much data yet. As a reference, PEGASUS was pre-trained with batch size 8192 for 1.5M steps.

Thank you. Are the steps correct for pretraining on another language?

**Comment by JingqingZ** [comment]
Yes, the steps looks correct

**Comment by argideritzalpea** [comment]
@EKebriaei Hi EKebriaei, when you say: "
@JingqingZ 
> 1- I implemented pretrain-parsing-ops.cc and did not use the code in the repo (Because of compilation errors).",

 are you saying that you wrote a custom script to do this operation? Did pretraining a new model end up working for you, and do you have the code to your implementation of `pretrain-parsing-ops`?
 
 It is not entirely clear how to train a new model on a new dataset.
